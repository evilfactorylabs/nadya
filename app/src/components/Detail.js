import React, { useState } from 'react'
import { navigate } from '@reach/router'
import { makeStyles } from '@material-ui/core/styles'
import {
  Card,
  CardContent,
  Typography,
  Menu,
  MenuItem,
  CardHeader,
  IconButton,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle
} from '@material-ui/core'

import MoreVertIcon from '@material-ui/icons/MoreVert'
import SimpleIcon from 'simple-icons-react-component'
import dayjs from 'dayjs'

import { getPeriod, toCurrency } from 'utils'

const useStyles = makeStyles({
  card: {
    textAlign: 'center',
    minWidth: 275
  },
  imageContainer: {
    width: 120,
    margin: 'auto',
    marginBottom: 20
  },
  pos: {
    marginBottom: 12
  }
})

const AlertDialog = ({
  isDialogOpen,
  handleClose,
  subscriptionTitle,
  handleDelete
}) => (
  <div>
    <Dialog open={isDialogOpen} onClose={handleClose}>
      <DialogTitle>Hapus {subscriptionTitle}?</DialogTitle>
      <DialogContent>
        <DialogContentText id='alert-dialog-description'>
          Apakah kamu yakin ingin menghapus subscription ini?
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color='primary' autoFocus>
          Kembali
        </Button>
        <Button onClick={handleDelete}>Hapus</Button>
      </DialogActions>
    </Dialog>
  </div>
)

export default ({ subscription, handleDelete }) => {
  const classes = useStyles()
  const [isDialogOpen, setDialogOpen] = useState(false)
  const [anchorEl, setAnchorEl] = React.useState(null)

  const handleClick = event => setAnchorEl(event.currentTarget)
  const handleClose = () => setAnchorEl(null)
  const handleEdit = () => navigate(`/${subscription._id}/edit`)
  const handleCloseDialog = () => setDialogOpen(false)

  const confirmDelete = () => {
    handleClose()
    setDialogOpen(true)
  }

  return (
    <Card className={classes.card} elevation={0}>
      <CardHeader
        action={
          <>
            <IconButton aria-label='Settings' onClick={handleClick}>
              <MoreVertIcon />
            </IconButton>
            <Menu
              id='simple-menu'
              anchorEl={anchorEl}
              keepMounted
              open={Boolean(anchorEl)}
              onClose={handleClose}
            >
              <MenuItem onClick={handleEdit}>Ubah</MenuItem>
              <MenuItem onClick={confirmDelete}>Hapus</MenuItem>
            </Menu>
          </>
        }
      />
      <CardContent>
        <div className={classes.imageContainer}>
          {subscription.title && <SimpleIcon name={subscription.title} />}
        </div>
        <Typography className={classes.pos} variant='h4'>
          {subscription.title}
        </Typography>
        <Typography>
          {toCurrency(subscription.cost)}/{getPeriod(subscription.period)}
        </Typography>
        <Typography variant='body2' component='p'>
          Sejak {dayjs(subscription.firstBill).format('DD MMMM YYYY')}
        </Typography>
      </CardContent>
      <AlertDialog
        subscriptionId={subscription._id}
        subscriptionTitle={subscription.title}
        handleDelete={handleDelete}
        handleClose={handleCloseDialog}
        isDialogOpen={isDialogOpen}
      />
    </Card>
  )
}
