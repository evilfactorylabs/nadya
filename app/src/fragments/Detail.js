import React, { useEffect, useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import CardHeader from '@material-ui/core/CardHeader'
import IconButton from '@material-ui/core/IconButton'
import MoreVertIcon from '@material-ui/icons/MoreVert'
import SimpleIcon from 'simple-icons-react-component'
import dayjs from 'dayjs'

import { navigate } from '@reach/router'
import { getPeriod, toCurrency } from '../utils'
import { getSubscription } from '../services/subscription'

import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'

import { deleteSubscription } from '../services/subscription'

function AlertDialog({
  isDialogOpen,
  handleClose,
  subscriptionId,
  subscriptionTitle
}) {
  const handleDelete = () => {
    deleteSubscription(subscriptionId).then(() => {
      navigate('/', { replace: true })
    })
  }

  return (
    <div>
      <Dialog open={isDialogOpen} onClose={handleClose}>
        <DialogTitle>Hapus {subscriptionTitle}?</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>
            Apakah kamu yakin ingin menghapus subscription ini?
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color='primary' autoFocus>
            Kembali
          </Button>
          <Button onClick={handleDelete}>Hapus</Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}

const useStyles = makeStyles({
  card: {
    textAlign: 'center',
    minWidth: 275
  },
  imageContainer: {
    width: 120,
    margin: 'auto',
    marginBottom: 20
  },
  pos: {
    marginBottom: 12
  }
})

export default ({ subscription_id }) => {
  const classes = useStyles()
  const [isDialogOpen, setDialogOpen] = useState(false)
  const [subscription, setSubscription] = useState({})
  const [anchorEl, setAnchorEl] = React.useState(null)

  const handleClick = event => setAnchorEl(event.currentTarget)
  const handleClose = () => setAnchorEl(null)

  const handleCloseDialog = () => setDialogOpen(false)
  const confirmDelete = () => {
    handleClose()
    setDialogOpen(true)
  }

  useEffect(() => {
    getSubscription(subscription_id).then(doc => {
      setSubscription(doc)
    })
  }, [subscription_id])

  return (
    <>
      {subscription && subscription.title && (
        <Card className={classes.card} elevation={0}>
          <CardHeader
            action={
              <>
                <IconButton aria-label='Settings' onClick={handleClick}>
                  <MoreVertIcon />
                </IconButton>
                <Menu
                  id='simple-menu'
                  anchorEl={anchorEl}
                  keepMounted
                  open={Boolean(anchorEl)}
                  onClose={handleClose}
                >
                  <MenuItem
                    onClick={() => navigate(`/${subscription._id}/edit`)}
                  >
                    Ubah
                  </MenuItem>
                  <MenuItem onClick={confirmDelete}>Hapus</MenuItem>
                </Menu>
              </>
            }
          />
          <CardContent>
            <div className={classes.imageContainer}>
              <SimpleIcon name={subscription.title} />
            </div>
            <Typography className={classes.pos} variant='h4'>
              {subscription.title}
            </Typography>
            <Typography>
              {toCurrency(subscription.cost)}/{getPeriod(subscription.period)}
            </Typography>
            <Typography variant='body2' component='p'>
              Sejak {dayjs(subscription.firstBill).format('DD MMMM YYYY')}
            </Typography>
          </CardContent>
          <AlertDialog
            subscriptionId={subscription._id}
            subscriptionTitle={subscription.title}
            handleClose={handleCloseDialog}
            isDialogOpen={isDialogOpen}
          />
        </Card>
      )}
    </>
  )
}
