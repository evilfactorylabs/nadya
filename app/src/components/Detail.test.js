import Detail from './Detail'

afterAll(async () => {
  await cleanup()
})

describe('Fragments: Detail.js', () => {
  test('It should render Detail screen', () => {
    const subscription = {
      title: 'Spotify'
    }
    const handleDelete = () => {}
    const { container } = render(
      <Detail handleDelete={handleDelete} subscription={subscription} />
    )

    expect(container).toBeInTheDocument()
  })
})
