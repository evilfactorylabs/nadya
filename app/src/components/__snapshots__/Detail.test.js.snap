// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragments: Detail.js It should render Detail screen 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      />
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-label="Settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <div
        class="makeStyles-imageContainer-2"
      >
        <div
          style="fill: currentColor; width: 100%; color: rgb(30, 215, 96);"
        >
          <svg
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Spotify icon
            </title>
            <path
              d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.66 0 12 0zm5.521 17.34c-.24.359-.66.48-1.021.24-2.82-1.74-6.36-2.101-10.561-1.141-.418.122-.779-.179-.899-.539-.12-.421.18-.78.54-.9 4.56-1.021 8.52-.6 11.64 1.32.42.18.479.659.301 1.02zm1.44-3.3c-.301.42-.841.6-1.262.3-3.239-1.98-8.159-2.58-11.939-1.38-.479.12-1.02-.12-1.14-.6-.12-.48.12-1.021.6-1.141C9.6 9.9 15 10.561 18.72 12.84c.361.181.54.78.241 1.2zm.12-3.36C15.24 8.4 8.82 8.16 5.16 9.301c-.6.179-1.2-.181-1.38-.721-.18-.601.18-1.2.72-1.381 4.26-1.26 11.28-1.02 15.721 1.621.539.3.719 1.02.419 1.56-.299.421-1.02.599-1.559.3z"
            />
          </svg>
        </div>
      </div>
      <h4
        class="MuiTypography-root makeStyles-pos-3 MuiTypography-h4"
      >
        Spotify
      </h4>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        IDRÂ 0
        /
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Sejak 
        10 August 2019
      </p>
    </div>
    <div />
  </div>
</div>
`;
